cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/EntityIdKeyHelper.java

```diff
@@ -140,7 +141,7 @@ public static NamespacedEntityId getTargetIdIdFromKey(MDSKey.Splitter keySplitte
       String name = keySplitter.getString();
       String version = keySplitter.getString();
       return new ArtifactId(namespaceId, name, version);
-    } else if (type.equals(TYPE_MAP.get(DatasetId.class))) {
+    } else if (type.equals(TYPE_MAP.get(DatasetId.class)) || type.equals(MetadataEntity.DATASET_INSTANCE)) {
```

Comment:
this should not be exposed in MetadataEntity. MetadataEntity is a publicly exposed API we should not expose deprecated stuff there.